app/src/main/java/com/github/ravifrancesco/softwaredevproject/POIPoint.java

```diff
@@ -28,11 +33,42 @@ public POIPoint(GeoPoint geoPoint) {
         super(geoPoint.getLatitude(), geoPoint.getLongitude(), geoPoint.getAltitude());
     }
 
-    public POIPoint(POI point) {
-        super(point.mLocation.getLatitude(), point.mLocation.getLongitude(), point.mLocation.getAltitude());
-        this.setName(point.mType);
+
+    /**
+     * Constructor for POI Point
+     * @param name peak name
+     * @param latitude peak latitude
+     * @param longitude peak longitude
+     * @param altitude peak height (in meters)
+     */
+    public POIPoint(String name, double latitude, double longitude, long altitude){
+        super(latitude, longitude, altitude);
+        this.setName(name);
+
     }
 
 
+    /**
+     * Override method for POIPoint HashSet comparison
+     * @return hashcode
+     */
+    @Override
+    public int hashCode() {
+        return name.hashCode();
+    }
 
+    /**
+     * Override method for POIPoint HashSet comparison
+     * Compare only the name of the POI
+     * @return
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) return false;
+        if (!(obj instanceof POIPoint))
+            return false;
+        if (obj == this)
+            return true;
+        return Objects.equals(this.getName(), ((POIPoint) obj).getName());
```

Comment:
```suggestion
        return getName().equals(((POIPoint)obj).getName());
```
As both are strings, maybe we can use the strings equals.